### PR TITLE
Moving workers into local apps. Adding concurrency.

### PR DESCRIPTION
**Pull request by geekforbrains**

### CODE REVIEW

**Comment by nerdburn** [comment]
Right on old chappy.

**Review by adriaanwm — APPROVED** [review]
Nice! Some notes:
- Would be helpful to have a built in way to stub tasks when writing tests (force sync, or just collect calls to tasks), with an example in Readme. 
- Tests for the workers app would be nice
- I'm wondering if the transaction.atomic applies to the function called by the pool.
- Might improve things to fetch tasks 20 + at a time instead of an sql query for each.

File: apps/workers/registry.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+
+def add(handler, schedule):
+    token = (handler, schedule)
+    if token not in _registry:
+        _registry.append(token)
+
+
+def get():
+    return _registry
```
[the remark this answers is not in the document]
Comment (geekforbrains) [reply apps/workers/registry.py:11]: I think there is a way in python 3 actually. Will look into. All the rest of your points are spot on! Thanks for diving into the details 👍 will prep an update.

File: apps/workerexample/tasks.py [hunk 2]
```diff
@@ -12,7 +12,7 @@ def say_hello(name):
     print('Hello', name)
 
 
-@task()
+#  @task()
```
Comment (eweisner1973) [inline apps/workerexample/tasks.py:15]: Is this supposed to be commented out?

**Review by eweisner1973 — APPROVED** [review]
Just a question about a commented task.

File: apps/workerexample/tasks.py (hunk 2 above)
Comment (geekforbrains) [reply apps/workerexample/tasks.py:15]: yea just while I was testing different variations of the test worker app

**Pull request merged**